client/src/services/SubtitleReader.js

```diff
@@ -44,6 +61,48 @@ export default class SubtitleReader {
             }));
         }
 
+        if (file.name.endsWith('.ytxml')) {
+            const xml = new window.DOMParser().parseFromString(await file.text(), 'text/xml');
+            const textNodes = xml.getElementsByTagName('text');
+            const subtitles = [];
+
+            for (let index = 0, length = textNodes.length; index < length; index++) {
+                const elm = textNodes[index];
+                const start = parseFloat(elm.getAttribute('start'));
+
+                subtitles.push({
+                    start: Math.floor(start * 1000),
+                    end: Math.floor((start + parseFloat(elm.getAttribute('dur'))) * 1000),
+                    text: this.decodeHTML(elm.textContent.replace(tagRegex, '')),
+                    track,
+                });
+            }
+
+            return subtitles;
+        }
+
         throw new Error('Unsupported subtitle file format');
     }
+
+    fixRTL(line) {
```

Comment:
It's currently conventional in this codebase to prefix private methods with an underscore.